sys-process/pidof-bsd: Fix build failure on FreeBSD 11.0.

nigoro opened the pull request:
Fix the following issue.
```
>>> Compiling source in /var/tmp/portage/sys-process/pidof-bsd-20050501-r4/work/pidof ...
make: "/usr/share/mk/bsd.opts.mk" line 97: warning: "NO_MAN is defined, but deprecated. Please use MK_MAN=no instead."
make: "/usr/share/mk/src.libnames.mk" line 395: /var/tmp/portage/sys-process/pidof-bsd-20050501-r4/work/pidof: These libraries should be LIBADD+=foo rather than DPADD/LDADD+=-lfoo:  kvm
 * ERROR: sys-process/pidof-bsd-20050501-r4::gentoo failed (compile phase):
```

mgorny commented on sys-process/pidof-bsd/pidof-bsd-20050501-r4.ebuild:
```diff
@@ -21,7 +22,13 @@ S="${WORKDIR}/pidof"
+src_prepare() {
+	epatch "${PATCHES[@]}"
+
+	append-ldflags -lkvm
```
I dunno about BSD build system but usually append `-l` to LDFLAGS is wrong since LDFLAGS are passed before objects.

nigoro commented:
Thanks for review.
I fixed the commit of this package.


```
make: "/usr/share/mk/src.libnames.mk" line 395: /var/tmp/portage/sys-process/pidof-bsd-20050501-r4/work/pidof: These libraries should be LIBADD+=foo rather than DPADD/LDADD+=-lfoo:  kvm
```
Cause, unnecessary files exists in /usr/share/mk.
I've updated freebsd-mk-defs and freebsd.eclass.
https://cgit.gentoo.org/proj/gentoo-bsd.git/commit/?id=0821460d6a35dd331a721b8ffe9b2017f32598ca

No longer need drop LDADD= -lkvm from pidof/Makefile.

aballier commented:
does it still build on fbsd 10 ?

(fbsd 11 is not in tree yet, 10 is still supported, so this cannot be merged if it breaks fbsd 10)

nigoro commented:
@aballier, Of course, you can build on 10.x and 9.x.
This fix will pass the error checking that has been enhanced with FreeBSD 11.0.

Error on FreeBSD 11.0.
```
>>> Compiling source in /var/tmp/portage/sys-process/pidof-bsd-20050501-r4/work/pidof ...
make: "/usr/share/mk/bsd.links.mk" line 17: Wrong number of words (1) in .for substitution list with 2 vars
make: Fatal errors encountered -- cannot continue
make: stopped in /var/tmp/portage/sys-process/pidof-bsd-20050501-r4/work/pidof
 * ERROR: sys-process/pidof-bsd-20050501-r4::gentoo failed (compile phase):
```

aballier commented:
then merged, thx!
